### PR TITLE
Apply flake8 to the numpy related files

### PR DESCRIPTION
**Pull request by esc**
This is a flake8 followup for #3900

### CODE REVIEW

File: numba/targets/arrayobj.py [hunk 1]
```diff
@@ -3208,9 +3220,10 @@ def make_array_nditer(context, builder, sig, args):
     res = nditer._getvalue()
     return impl_ret_borrowed(context, builder, nditerty, res)
 
+
 @lower_builtin('iternext', types.NumpyNdIterType)
 @iternext_impl(RefType.BORROWED)
-def iternext_numpy_ndindex(context, builder, sig, args, result):
+def iternext_numpy_ndindex_borrowed(context, builder, sig, args, result):
```
Comment (stuartarchibald) [inline numba/targets/arrayobj.py]: all the refs are borrowed, the difference here is that this is an iterator for the `nditer` type, whereas the above is for the `ndindex` type.

File: numba/targets/arrayobj.py [hunk 2]
```diff
@@ -2827,7 +2837,7 @@ def _make_flattening_iter_cls(flatiterty, kind):
     assert kind in ('flat', 'ndenumerate')
 
     array_type = flatiterty.array_type
-    dtype = array_type.dtype
+    _ = array_type.dtype
```
Comment (stuartarchibald) [inline numba/targets/arrayobj.py]: this is dead?

File: numba/targets/arrayobj.py [hunk 3]
```diff
@@ -2861,7 +2871,7 @@ def init_specific(self, context, builder, arrty, arr):
             # where the strides are unknown at compile-time.
 
             def iternext_specific(self, context, builder, arrty, arr, result):
-                zero = context.get_constant(types.intp, 0)
+                _ = context.get_constant(types.intp, 0)
```
Comment (stuartarchibald) [inline numba/targets/arrayobj.py]: this is dead?

File: numba/targets/arrayobj.py [hunk 4]
```diff
@@ -2941,7 +2951,7 @@ def init_specific(self, context, builder, arrty, arr):
 
             def iternext_specific(self, context, builder, arrty, arr, result):
                 ndim = arrty.ndim
-                data = arr.data
+                _ = arr.data
```
Comment (stuartarchibald) [inline numba/targets/arrayobj.py]: this is dead?

File: numba/targets/arrayobj.py [hunk 5]
```diff
@@ -3092,12 +3102,12 @@ def iternext_numpy_getitem(context, builder, sig, args):
     arrcls = context.make_array(arrty)
     arr = arrcls(context, builder, value=flatiter.array)
 
-    res = flatiter.setitem(context, builder, arrty, arr, index, value)
+    _ = flatiter.setitem(context, builder, arrty, arr, index, value)
```
Comment (stuartarchibald) [inline numba/targets/arrayobj.py]: assignment can be removed?

File: numba/targets/arrayobj.py [hunk 6]
```diff
@@ -3292,8 +3306,8 @@ def _parse_shape(context, builder, ty, val):
     else:
         assert isinstance(ty, types.BaseTuple)
         ndim = ty.count
-        arrshape = context.cast(builder, val, ty,
-                                types.UniTuple(types.intp, ndim))
+        _ = context.cast(builder, val, ty,
```
Comment (stuartarchibald) [inline numba/targets/arrayobj.py]: think this is dead.

File: numba/targets/arrayobj.py [hunk 7]
```diff
@@ -3565,8 +3585,8 @@ def diag_impl(arr, k=0):
         def diag_impl(arr, k=0):
             #Will return arr.diagonal(v, k) when axis args are supported
             rows, cols = arr.shape
-            r = rows
-            c = cols
+            _ = rows
+            _ = cols
```
Comment (stuartarchibald) [inline numba/targets/arrayobj.py]: both are dead

File: numba/targets/arrayobj.py [hunk 8]
```diff
@@ -4038,7 +4068,7 @@ def check_sequence_shape(context, builder, seqty, seq, shapes):
     """
     Check the nested sequence matches the given *shapes*.
     """
-    intp_t = context.get_value_type(types.intp)
+    _ = context.get_value_type(types.intp)
```
Comment (stuartarchibald) [inline numba/targets/arrayobj.py]: dead

File: numba/targets/arrayobj.py [hunk 9]
```diff
@@ -4377,7 +4409,7 @@ def _do_concatenate(context, builder, axis,
 def _np_concatenate(context, builder, arrtys, arrs, retty, axis):
     ndim = retty.ndim
 
-    zero = cgutils.intp_t(0)
+    _ = cgutils.intp_t(0)
```
Comment (stuartarchibald) [inline numba/targets/arrayobj.py]: dead

File: numba/targets/arrayobj.py [hunk 10]
```diff
@@ -4430,7 +4462,7 @@ def _np_stack(context, builder, arrtys, arrs, retty, axis):
 
     zero = cgutils.intp_t(0)
     one = cgutils.intp_t(1)
-    ll_ndim = cgutils.intp_t(ndim)
+    _ = cgutils.intp_t(ndim)
```
Comment (stuartarchibald) [inline numba/targets/arrayobj.py]: dead

File: numba/tests/test_np_functions.py [hunk 11]
```diff
@@ -253,10 +290,23 @@ def check(x_types, x_values, **kwargs):
         check(x_types, x_values)
 
         # complex domain scalar context
-        x_values = [1.+0j, -1+0j, 0.0+0.0j, -0.0+0.0j, 0+1j, 0-1j, 0.5+0.0j,
-                    -0.5+0.0j, 0.5+0.5j, -0.5-0.5j, 5+5j, -5-5j,
+        x_values = [1. + 0j,
```
Comment (stuartarchibald) [inline numba/tests/test_np_functions.py]: think this reads worse than before :) perhaps put it back and `#noqa` ?

File: numba/tests/test_np_functions.py [hunk 12]
```diff
@@ -295,8 +345,18 @@ def check(x_types, x_values):
         check(x_types, x_values)
 
         # complex domain scalar context
-        x_values = [1.+0j, -1+0j, 0.0+0.0j, -0.0+0.0j, 1j, -1j, 0.5+0.0j,
-                    -0.5+0.0j, 0.5+0.5j, -0.5-0.5j, 5+5j, -5-5j]
+        x_values = [1. + 0j,
```
Comment (stuartarchibald) [inline numba/tests/test_np_functions.py]: same, put it back and #noqa ?

**Comment by esc** [comment]
Indeed all the places where I inserted `_` the assigned variable wasn't being used. This is one way of shutting up flake8 in such cases. The other is to remove the presumably dead-code. Issue with the later option is that code might have a side-effect which then causes breakage in the subsequent code as the side-effect no longer happened. What I'll do, is try to remove all the dead code and see what the CI system says.

**Comment by esc** [comment]
So, let's see if I inadvertently destroyed anything.

File: numba/targets/arrayobj.py (hunk 5 above)
Comment (stuartarchibald) [reply numba/targets/arrayobj.py]: @esc note the *assignment* can be removed, not the call itself. https://github.com/numba/numba/pull/3927/files#diff-81d8f85ead1fb3ae948c94ed8d2a969eL3095, `setitem` has a side effect

**Comment by esc** [comment]
@stuartarchibald this thing finally passed!

**Comment by stuartarchibald** [comment]
@esc thanks for the patch. Looks good. Marking for merge.

**Comment by esc** [comment]
🎉

**Comment by stuartarchibald** [comment]
Actually, given this file might cause merge conflicts I'm going to mark as waiting on reviewer, i.e. me, so it gets merged at a minimally conflicting time.

**Comment by esc** [comment]
Sure, no problem, I can rebase anytime if need be.

**Comment by esc** [comment]
rebased

**Comment by stuartarchibald** [comment]
NOTE: I've re-reviewed after the rebase, looks fine.

**Pull request merged**